api/src/pcapi/routes/backoffice/finance/finance_incidents_blueprint.py

```diff
@@ -54,18 +57,107 @@ def _get_incidents() -> list[finance_models.FinanceIncident]:
         .load_only(offerer_models.Offerer.id, offerer_models.Offerer.name),
+        sa.orm.joinedload(
+            finance_models.BookingFinanceIncident, finance_models.FinanceIncident.booking_finance_incidents
+        )
+        .joinedload(finance_models.BookingFinanceIncident.collectiveBooking)
+        .joinedload(educational_models.CollectiveBooking.finance_events),
     )
+
+    if form.status.data:
+        query = query.filter(finance_models.FinanceIncident.status.in_(form.status.data))
+
+    if form.offerer.data:
+        query = query.join(finance_models.FinanceIncident.venue).filter(
+            offerer_models.Venue.managingOffererId.in_(form.offerer.data)
+        )
+
+    if form.from_date.data or form.to_date.data:
+        query = query.join(history_models.ActionHistory, finance_models.FinanceIncident.action_history)
+        query = query.options(
+            sa.orm.joinedload(
+                history_models.ActionHistory,
+                finance_models.FinanceIncident.action_history,
+            )
+        )
+
+        if form.from_date.data:
+            from_datetime = date_utils.date_to_localized_datetime(form.from_date.data, datetime.min.time())
+            query = query.filter(history_models.ActionHistory.actionDate >= from_datetime)
+
+        if form.to_date.data:
+            to_datetime = date_utils.date_to_localized_datetime(form.to_date.data, datetime.max.time())
+            query = query.filter(history_models.ActionHistory.actionDate <= to_datetime)
+
+    if form.q.data:
+        search_query = form.q.data
+        if search_query.isnumeric():
+            query = query.options(
+                sa.orm.joinedload(
+                    finance_models.BookingFinanceIncident, finance_models.FinanceIncident.booking_finance_incidents
+                ).joinedload(finance_models.BookingFinanceIncident.finance_events)
+            )
+            query = query.join(
+                finance_models.BookingFinanceIncident, finance_models.FinanceIncident.booking_finance_incidents
+            )
+            booking_query = query.join(bookings_models.Booking, finance_models.BookingFinanceIncident.booking)
+            length_token = 6
+            queries = []
+            if len(form.q.data) == length_token:
+                queries.append(booking_query.filter(bookings_models.Booking.token == form.q.data))
+            query_id = query.filter(finance_models.FinanceIncident.id == form.q.data)
+            individual_query = booking_query.filter(bookings_models.Booking.id == form.q.data)
+            collective_query = query.join(
+                educational_models.CollectiveBooking, finance_models.BookingFinanceIncident.collectiveBooking
+            ).filter(educational_models.CollectiveBooking.id == form.q.data)
+
+            queries.append(query_id)
+            queries.append(individual_query)
+            queries.append(collective_query)
+            query = queries[0].union(*queries[1:])
```

Comment:
Peu lisible... Je propose de réorganiser par exemple ainsi pour que la lecture soit plus compréhensible : 

```suggestion
            query_id = query.filter(finance_models.FinanceIncident.id == int(form.q.data))

            if len(form.q.data) == TOKEN_LENGTH:
                individual_query = booking_query.filter(
                    sa.or_(
                        bookings_models.Booking.id == int(form.q.data), 
                        bookings_models.Booking.token == form.q.data
                    )
                )
            else:
                individual_query = booking_query.filter(bookings_models.Booking.id == int(form.q.data))
                
            collective_query = query.join(
                educational_models.CollectiveBooking, finance_models.BookingFinanceIncident.collectiveBooking
            ).filter(educational_models.CollectiveBooking.id == int(form.q.data))
            
            query = query_id.union(individual_query, collective_query)
```